Remotion/Validation/UnitTests/RuleBuilders/AddingObjectValidationRuleBuilderTest.cs

```diff
@@ -14,61 +14,153 @@
+    private Mock<IObjectMetaValidationRuleCollector> _objectMetaValidationRuleCollectorMock;
 
+    [SetUp]
+    public void SetUp ()
+    {
+      _addingObjectValidationRuleCollectorMock = new Mock<IAddingObjectValidationRuleCollector>(MockBehavior.Strict);
+      _addingObjectValidationRuleCollectorMock
+          .Setup(stub => stub.ValidatedType)
+          .Returns(TypeAdapter.Create(typeof(Customer)));
+
+      _objectMetaValidationRuleCollectorMock = new Mock<IObjectMetaValidationRuleCollector>(MockBehavior.Strict);
+      _addingObjectValidationRuleCollectorMock
+          .Setup(stub => stub.ValidatedType)
+          .Returns(TypeAdapter.Create(typeof(Customer)));
+
+      _addingObjectValidationBuilder = new AddingObjectValidationRuleBuilder<Customer>(
+          _addingObjectValidationRuleCollectorMock.Object,
+          _objectMetaValidationRuleCollectorMock.Object);
+
+      _objectValidatorStub = new Mock<IObjectValidator>();
+    }
 
     [Test]
-    [Ignore("TODO RM-5906")]
     public void Initialization ()
     {
+      Assert.That(_addingObjectValidationBuilder.AddingObjectValidationRuleCollector, Is.SameAs(_addingObjectValidationRuleCollectorMock.Object));
+      Assert.That(_addingObjectValidationBuilder.ObjectMetaValidationRuleCollector, Is.SameAs(_objectMetaValidationRuleCollectorMock.Object));
     }
 
     [Test]
-    [Ignore("TODO RM-5906")]
     public void SetValidator ()
     {
+      Func<ObjectValidationRuleInitializationParameters, IObjectValidator> validatorFactory = _ => _objectValidatorStub.Object;
+      _addingObjectValidationRuleCollectorMock
+          .Setup(mock => mock.RegisterValidator(validatorFactory))
+          .Verifiable();
+
+      ((IAddingObjectValidationRuleBuilder<Customer>)_addingObjectValidationBuilder).SetValidator(validatorFactory);
+
+      _addingObjectValidationRuleCollectorMock.Verify();
     }
 
     [Test]
-    [Ignore("TODO RM-5906")]
     public void CanBeRemoved ()
     {
+      _addingObjectValidationRuleCollectorMock
+          .Setup(mock => mock.SetRemovable())
+          .Verifiable();
+
+      _addingObjectValidationBuilder.CanBeRemoved();
+
+      _addingObjectValidationRuleCollectorMock.Verify();
     }
 
     [Test]
-    [Ignore("TODO RM-5906")]
     public void AddMetaValidationRule ()
     {
+      var metaValidationRuleStub = new Mock<IObjectMetaValidationRule>();
+
+      _objectMetaValidationRuleCollectorMock
+          .Setup(mock => mock.RegisterMetaValidationRule(metaValidationRuleStub.Object))
+          .Verifiable();
+
+      _addingObjectValidationBuilder.AddMetaValidationRule(metaValidationRuleStub.Object);
+
+      _objectMetaValidationRuleCollectorMock.Verify();
     }
 
     [Test]
-    [Ignore("TODO RM-5906")]
     public void AddMetaValidationRule_FuncOverload ()
     {
+      var fakeValidationResult = MetaValidationRuleValidationResult.CreateValidResult();
+
+      IObjectMetaValidationRule rule = null;
```

Comment:
naming: I prefer to use "actual" in the name, e.g. "actualRule" to identify the value we want to check. Or "result", when it's the return value of the exercised SUT.